ambry-store/src/test/java/com.github.ambry.store/IndexSegmentTest.java

```diff
@@ -92,14 +116,19 @@ public void cleanup() throws IOException {
    * @throws StoreException
    */
   @Test
-  public void comprehensiveTest() throws IOException, StoreException {
+  public void comprehensiveTest() throws IOException, StoreException, InterruptedException {
     String[] logSegmentNames = {LogSegmentNameHelper.generateFirstSegmentName(false), generateRandomLogSegmentName()};
     for (String logSegmentName : logSegmentNames) {
       long writeStartOffset = Utils.getRandomLong(TestUtils.RANDOM, 1000);
       Offset startOffset = new Offset(logSegmentName, writeStartOffset);
       NavigableMap<MockId, IndexValue> referenceIndex = new TreeMap<>();
+      // advance time so that last modified time for VERSION_1 has different last modified times for different index
+      // segments
+      UtilsTest.advanceTime(time, 10000);
       IndexSegment indexSegment = generateIndexSegment(startOffset);
-      verifyIndexSegmentDetails(indexSegment, startOffset, 0, false, startOffset.getOffset());
+      verifyIndexSegmentDetails(indexSegment, startOffset, 0, false, startOffset.getOffset(), false, time.seconds(),
+          null);
+      resetKey = indexSegment.getResetKey();
```

Comment:
this is misleading. `indexSegment.getResetKey()` is `null` at this point - why do this at all? It is `addPutEntries()` that actually sets up the resetKey.